src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,118 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
```

Comment:
The `Node<T>` class should be an inner class within `MyLinkedList` to improve code readability and maintainability as per the checklist.